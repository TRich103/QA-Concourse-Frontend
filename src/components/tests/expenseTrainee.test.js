import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme,{shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

import ExpenseTrainee from '../expenses-trainee.component';

Enzyme.configure({adapter: new Adapter()})

let component;
let props = {match:{params:{id:'5d0bb39bd2ba63099c621593'}}};

beforeEach(() => {
    component = shallow(<ExpenseTrainee {...props}/>);
});


describe('render testing', () => {
    it('renders correctly when there are no items', () => {
        // resolve, status, id
        expect(toJson(component)).toMatchSnapshot();
    });

    it('checking if labels are correctly rendered if other is not selected', () => {
        expect(component.find('label')).toHaveLength(2);
    });

    it('checking if additional label is correctly rendered', () => {
        component.setState({
            other: true
        });
        expect(component.find('label')).toHaveLength(3);
    });

    it('checking if add and back buttons are rendered', () => {
        expect(component.find('Button')).toHaveLength(2);
    });

    // it('checking if additional label is correctly rendered', () => {
    //     const clickCallback = sinon.spy();
    //     const actualNode = shallow(<ExpenseTrainee {...props} onSave={clickCallback} />);
    //     actualNode.setState({
    //         other: true
    //     })
    //     // WHEN
    //     actualNode.find("#createExpenseBtn").at(0).simulate("click");
    //     // THEN
    //     sinon.assert.called(clickCallback);
    // });
})