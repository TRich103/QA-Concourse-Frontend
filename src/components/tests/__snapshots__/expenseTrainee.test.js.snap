// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render testing renders correctly when there are no items 1`] = `
<div
  className="BigBox"
>
  <div
    className="QAtable"
  >
    <div>
      <h2>
        Expenses for 
         - 
      </h2>
      <hr />
      <div>
           
        <label>
          Amount (£): 
        </label>
          
        <input
          min="0"
          onChange={[Function]}
          style={
            Object {
              "marginBlock": 30,
            }
          }
          type="number"
          value={0}
        />
             
        <label>
          Type: 
        </label>
          
        <select
          id="expenseType"
          onChange={[Function]}
          value=""
        >
          <option
            selected={true}
            value=""
          >
            Select expense type
          </option>
          <option
            value="Air fares"
          >
            Air fares
          </option>
          <option
            value="Car hire"
          >
            Car hire
          </option>
          <option
            value="Computer consumables"
          >
            Computer consumables
          </option>
          <option
            value="Customer Entertainment(customer)"
          >
            Customer entertainment(customer)
          </option>
          <option
            value="Customer Entertainment(staff)"
          >
            Customer entertainment(staff)
          </option>
          <option
            value="Hotels/Accomodation"
          >
            Hotels/Accomodation
          </option>
          <option
            value="Meals"
          >
            Meals
          </option>
          <option
            value="Fuel"
          >
            Fuel
          </option>
          <option
            value="Other"
          >
            Other
          </option>
          <option
            value="Parking/Tolls"
          >
            Parking/Tolls
          </option>
          <option
            value="Postage"
          >
            Postage
          </option>
          <option
            value="Public Transport"
          >
            Public Transport
          </option>
          <option
            value="Staff Entertainment"
          >
            Staff Entertainment
          </option>
          <option
            value="Stationary"
          >
            Stationary
          </option>
          <option
            value="Subscription and req. fees"
          >
            Subscription and req. fees
          </option>
          <option
            value="Taxi Fares"
          >
            Taxi Fares
          </option>
          <option
            value="Training Materials"
          >
            Training Materials
          </option>
        </select>
             
        <ButtonGroup
          role="group"
          tag="div"
        >
            
          <Button
            color="secondary"
            id="createExpenseBtn"
            onClick={[Function]}
            tag="button"
            type="submit"
          >
            Add
          </Button>
           
        </ButtonGroup>
        <table
          className="table table-striped"
          id="logTable"
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <thead>
            <tr>
              <th>
                Expenses
              </th>
              <th>
                Amount(£)
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
           
      <ButtonGroup
        role="group"
        tag="div"
      >
        <Button
          color="secondary"
          id="createExpenseBtn"
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 30,
            }
          }
          tag="button"
          type="submit"
        >
          ⭠ Back
        </Button>
      </ButtonGroup>
    </div>
  </div>
</div>
`;
